plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE NamedFieldPuns    #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications  #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Ledger.Tx
+    ( module Export
+    -- * Transactions
+    , addSignature
+    , pubKeyTxOut
+    , scriptTxOut
+    , scriptTxOut'
+    , updateUtxo
+    , txOutRefs
+    , unspentOutputsTx
+    -- ** Hashing transactions
+    , txId
+    ) where
+
+import           Cardano.Crypto.Hash       (SHA256, digest)
+import qualified Codec.CBOR.Write          as Write
+import           Codec.Serialise.Class     (encode)
+import           Control.Lens
+import           Data.Map                  (Map)
+import qualified Data.Map                  as Map
+import           Data.Proxy
+import qualified Data.Set                  as Set
+import           Data.Text.Prettyprint.Doc (Pretty (pretty), braces, colon, hang, nest, viaShow, vsep, (<+>))
+import           Ledger.Address            (pubKeyAddress, scriptAddress)
+import           Ledger.Crypto             (PrivateKey, PubKey, signTx, toPublicKey)
+import           Ledger.Scripts            (Datum, Validator, datumHash)
+import           Plutus.V1.Ledger.Tx       as Export
+import           Plutus.V1.Ledger.TxId     (TxId (..))
+import           Plutus.V1.Ledger.Value
+
+instance Pretty Tx where
+    pretty t@Tx{txInputs, txCollateral, txOutputs, txMint, txFee, txValidRange, txSignatures, txMintScripts, txData} =
+        let lines' =
+                [ hang 2 (vsep ("inputs:" : fmap pretty (Set.toList txInputs)))
+                , hang 2 (vsep ("collateral inputs:" : fmap pretty (Set.toList txCollateral)))
+                , hang 2 (vsep ("outputs:" : fmap pretty txOutputs))
+                , "mint:" <+> pretty txMint
+                , "fee:" <+> pretty txFee
+                , hang 2 (vsep ("mps:": fmap pretty (Set.toList txMintScripts)))
+                , hang 2 (vsep ("signatures:": fmap (pretty . fst) (Map.toList txSignatures)))
+                , "validity range:" <+> viaShow txValidRange
+                , hang 2 (vsep ("data:": fmap (pretty . snd) (Map.toList txData) ))
+                ]
+            txid = txId t
+        in nest 2 $ vsep ["Tx" <+> pretty txid <> colon, braces (vsep lines')]
+
+-- | Compute the id of a transaction.
+txId :: Tx -> TxId
+-- Double hash of a transaction, excluding its witnesses.
+txId tx = TxId $ digest (Proxy @SHA256)
+               $ digest (Proxy @SHA256)
+               (Write.toStrictByteString $ encode $ strip tx)
```

Comment:
@michaelpj I reimplemented the function with `cardano-crypto-class` instead of `cryptonite`. Does this seem to you like the same behavior as the last implementation. As reference, here's the old implementation:

```
-- | Compute the id of a transaction.
txId :: Tx -> TxId
-- Double hash of a transaction, excluding its witnesses.
txId tx = TxId $ BA.convert h' where
    h :: Digest SHA256
    h = hash $ Write.toStrictByteString $ encode $ strip tx
    h' :: Digest SHA256
    h' = hash h
```

Reply:
Seems okay to me. This should also die in the medium term as we move to using `cardano-api` transactions more.